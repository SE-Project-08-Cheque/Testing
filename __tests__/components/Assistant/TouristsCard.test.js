// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";



// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import TouristCard from "../../../components/Assistant/TouristsCard";


afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
    ReactDOM.render(        
            <TouristCard />        
        , div);
});

it("renders tourists card correctly", () => {
    const { getByTestId } = render(
        
        <TouristCard
                            key="01"
                            id="01"
                            imageUrl="https://upload.wikimedia.org/wikipedia/commons/d/d3/User_Circle.png"
                            first_name="Chris"
                            last_name="Haynes"
                            email="Chrishaynes@gmail.com"
                            phone="0771234567"
                
            />
        
    );
    expect(getByTestId('full_name')).toHaveTextContent("Chris Haynes");
    expect(getByTestId('email')).toHaveTextContent("Chrishaynes@gmail.com");
    expect(getByTestId('phone')).toHaveTextContent("0771234567");

});

it("matches snapshot", () => {
    const tree = renderer.create(
        <TouristCard
            key="01"
            id="01"
            imageUrl="https://upload.wikimedia.org/wikipedia/commons/d/d3/User_Circle.png"
            first_name="Chris"
            last_name="Haynes"
            email="Chrishaynes@gmail.com"
            phone="0771234567"
  />);

    expect(tree).toMatchSnapshot();
})
