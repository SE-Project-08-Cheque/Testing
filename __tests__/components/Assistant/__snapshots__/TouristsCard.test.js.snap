// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-zn4p7q"
>
  <div
    className="css-19fmqat"
  >
    <div
      className="css-1txosv4"
    >
      <div
        className="css-gmuwbf"
      >
        <img
          className="chakra-image__placeholder css-1kdrupq"
        />
      </div>
    </div>
    <div
      className="css-1h9vd2j"
    >
      <div
        className="css-n16x53"
      >
        <div
          className="css-x1sij0"
        >
          <span
            className="chakra-badge css-637omx"
          >
            New
          </span>
        </div>
        <h4
          className="css-1mwm0bw"
          data-testid="full_name"
        >
          Chris Haynes
        </h4>
        <div
          className="css-0"
          data-testid="email"
        >
          Chrishaynes@gmail.com
        </div>
        <div
          className="css-0"
          data-testid="phone"
        >
          0771234567
        </div>
      </div>
    </div>
    <div
      className="css-1xeh2fo"
    >
      <div
        className="chakra-wrap css-0"
      >
        <ul
          className="chakra-wrap__list css-yp34du"
        >
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <button
              className="chakra-button css-oqfhsh"
              type="button"
            >
              <span
                className="chakra-button__icon css-ktvjz1"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.49 5.48c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-3.6 13.9l1-4.4 2.1 2v6h2v-7.5l-2.1-2 .6-3c1.3 1.5 3.3 2.5 5.5 2.5v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1l-5.2 2.2v4.7h2v-3.4l1.8-.7-1.6 8.1-4.9-1-.4 2 7 1.4z"
                  />
                </svg>
              </span>
              Activities
            </button>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <button
              className="chakra-button css-oqfhsh"
              onClick={[Function]}
              type="button"
            >
              <span
                className="chakra-button__icon css-ktvjz1"
              >
                <svg
                  aria-hidden={true}
                  className="chakra-icon css-onkibi"
                  focusable={false}
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeWidth="2"
                  >
                    <path
                      d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                    />
                    <path
                      d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                    />
                  </g>
                </svg>
              </span>
              Update Details
            </button>
          </li>
          <li
            className="chakra-wrap__listitem css-zji64q"
          >
            <button
              className="chakra-button css-oqfhsh"
              onClick={[Function]}
              type="button"
            >
              <span
                className="chakra-button__icon css-ktvjz1"
              >
                <svg
                  aria-hidden={true}
                  className="chakra-icon css-onkibi"
                  focusable={false}
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                    />
                  </g>
                </svg>
              </span>
              Delete
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
